index.js

```diff
@@ -129,6 +207,10 @@ class Conf {
 			throw new TypeError('Use `delete()` to clear values');
 		}
 
+		if (typeof key === 'string' && key.indexOf(INTERNAL_KEY) === 0) {
```

Comment:
`key.indexOf(INTERNAL_KEY) === 0` will match even `__internal__foo` which is not something people would do, probably, but it's still an incorrect check.

Reply:
Makes sense! Updated it and added 2 tests to check your case and when `accessPropertiesByDotNotation` is set to false